internal/hcsoci/resources_wcow.go

```diff
@@ -138,5 +138,24 @@ func allocateWindowsResources(ctx context.Context, coi *createOptionsInternal, r
 		}
 	}
 
+	if cs, ok := coi.Spec.Windows.CredentialSpec.(string); ok {
+		// Only need to create a CCG instance for v2 containers
+		if schemaversion.IsV21(coi.actualSchemaVersion) {
+			hypervisorIsolated := false
+			if coi.HostingSystem != nil {
+				hypervisorIsolated = true
+			}
+			ccgState, err := CreateCredentialGuard(ctx, coi.actualID, cs, hypervisorIsolated)
```

Comment:
`CreateCredentialGuard` should return the resource object directly, perhaps as another return value?